pallets/moonbeam-foreign-assets/src/lib.rs

```diff
@@ -364,6 +367,27 @@ pub mod pallet {
 			.map_err(Into::into)
 		}
 
+		/// Transfer an asset from an account to another one
+		pub fn transfer(
+			asset_id: AssetId,
+			from: T::AccountId,
+			to: T::AccountId,
+			amount: U256,
+		) -> Result<(), evm::EvmError> {
+			EvmCaller::<T>::erc20_transfer(
+				Self::contract_address_from_asset_id(asset_id),
+				T::AccountIdToH160::convert(from),
+				T::AccountIdToH160::convert(to),
+				amount,
+			)
```

Comment:
Since https://github.com/moonbeam-foundation/moonbeam/pull/3258 was reverted, we may need to wrap it inside `with_storage_layer`:
```suggestion
			frame_support::storage::with_storage_layer(|| {
				EvmCaller::<T>::erc20_transfer(
					Self::contract_address_from_asset_id(asset_id),
					T::AccountIdToH160::convert(from),
					T::AccountIdToH160::convert(to),
					amount,
				)
			})
```